src/XMakeTasks/StateFileBase.cs

```diff
@@ -17,13 +17,12 @@ namespace Microsoft.Build.Tasks
     [Serializable()]
     internal class StateFileBase
     {
-        /// <summary>
-        /// Default constructor
-        /// </summary>
-        internal StateFileBase()
-        {
-            // do nothing
-        }
+        // Current version for serialization. This should be changed when breaking changes
+        // are made to this class.
+        private const byte CurrentSerializationVersion = 1;
```

Comment:
Would definitely be nice for this to be more automatic, maybe hash-of-tools like @dannyvv suggested.  File an up-for-grabs issue for future improvement?